docs/executor/kubernetes.rst

```diff
@@ -34,3 +34,71 @@ The volumes are optional and depend on your configuration. There are two volumes
   - By storing logs onto a persistent disk, the files are accessible by workers and the webserver. If you don't configure this, the logs will be lost after the worker pods shuts down
 
   - Another option is to use S3/GCS/etc to store logs
+
+KubernetesExecutor Architecture
+################################
+
+The KubernetesExecutor runs as a single pod that only requires access to the Kubernetes API. There are no external brokers or consistent workers needed.
```

Comment:
It is possible to run KubeExecutor from outside of a cluster too. Perhaps we can ignore that here though.

Also the KubeExecutor doesn't run as a pod, but runs inside the scheduler.

No external brokers, but still need the (existing) metadata DB, which can't be sqlite (let's be explicit.)

Reply:
These are all good points. Will fix now.